lib/rules/prefer-interface.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * @fileoverview Prefer an interface declaration over a type literal (type T = { ... })
+ * @author Armano <https://github.com/armano2>
+ */
+"use strict";
+
+const util = require("../util");
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description:
+                "Prefer an interface declaration over a type literal (type T = { ... })",
+            extraDescription: [util.tslintRule("interface-over-type-literal")],
+            category: "TypeScript",
+            url: util.metaDocsUrl("prefer-interface"),
+        },
+        fixable: "code",
+        messages: {
+            interfaceOverType: "Use an interface instead of a type literal.",
+        },
+        schema: [],
+    },
+    create(context) {
+        const sourceCode = context.getSourceCode();
+
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+        return {
+            // VariableDeclaration with kind type has only one VariableDeclarator
+            "VariableDeclaration[kind='type'] > VariableDeclarator[init.type='TSTypeLiteral']"(
+                node
+            ) {
+                context.report({
+                    node,
+                    messageId: "interfaceOverType",
+                    fix(fixer) {
```

Comment:
Do you think this would be easier to read as a generator?

```js
function fix*(fixer) {
  const typeNode = node.typeParameters || node.id;
  yield fixer.replaceText(sourceCode.getFirstToken(node.parent), "interface");
  yield fixer.replaceTextRange([typeNode.range[1], node.init.range[0]], " ");
  const afterToken = sourceCode.getTokenAfter(node.init);
  if (
    afterToken &&
    afterToken.type === "Punctuator" &&
    afterToken.value === ";"
  ) {
    yield fixer.remove(afterToken);
  }
}
```

Reply:
its not going to work, `eslint-plugin/fixer-return` is not allowing us to use generator

Reply:
(⏩ https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/issues/74)